internal/mode/static/nginx/config/upstreams.go

```diff
@@ -35,6 +36,49 @@ func (g GeneratorImpl) executeUpstreams(conf dataplane.Configuration) []executeR
 	return []executeResult{result}
 }
 
+func (g GeneratorImpl) executeStreamUpstreams(conf dataplane.Configuration) []executeResult {
+	upstreams := g.createStreamUpstreams(conf.StreamUpstreams)
+
+	result := executeResult{
+		dest: streamConfigFile,
+		data: helpers.MustExecuteTemplate(upstreamsTemplate, upstreams),
+	}
+
+	return []executeResult{result}
+}
+
+func (g GeneratorImpl) createStreamUpstreams(upstreams []dataplane.Upstream) []stream.Upstream {
+	ups := make([]stream.Upstream, 0, len(upstreams))
+
+	for _, u := range upstreams {
+		if len(u.Endpoints) != 0 {
+			ups = append(ups, g.createStreamUpstream(u))
+		}
+	}
+
+	return ups
+}
+
+func (g GeneratorImpl) createStreamUpstream(up dataplane.Upstream) stream.Upstream {
+	zoneSize := ossZoneSize
```

Comment:
zone sizes might be different for stream upstreams. I suggest calculating them for steam and http upstreams separately

Reply:
Maybe create stream zone size variables and utilize those? I could leave them the same as http for now

Reply:
How did you calculate the numbers for HTTP upstreams?

Reply:
NGINX fails to reload with it cannot fit configured upstream servers. So basically, adding upstream servers until NGINX fails - that shows how many upstream servers a zone can hold

Reply:
64 and 128k
576 and 512k
1248 and 1m

These are the results I got NGINX OSS. Should I just leave the same zone size for OSS in stream?

Reply:
for N+ 

111 and 128k
239 and 512k
991 and 1m

I think we can just leave this at 1m for nginx plus

Reply:
@sarthyparty  thanks. 

> These are the results I got NGINX OSS. Should I just leave the same zone size for OSS in stream?
👍 

> I think we can just leave this at 1m for nginx plus
👍 

could you update this comment with your results for stream upstreams?

https://github.com/nginxinc/nginx-gateway-fabric/blob/8f5b9e688cc54b4784d2113b9871bafc55987329/internal/mode/static/nginx/config/upstreams_template.go#L3-L6

Reply:
yup